src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Constructor.cs

```diff
@@ -0,0 +1,1365 @@
+    {
+        var source = """
+            using System.Collections.Generic;
+            
+            class MyList<T> : List<T> where T : class
+            {
+                public MyList(T item) : base() { }
+            }
+            
+            class C
+            {
+                void M()
+                {
+                    MyList<int> list = [with(42)];
+                }
+            }
+            """;
+
+        CreateCompilation(source).VerifyDiagnostics(
+            // (12,16): error CS0452: The type 'int' must be a reference type in order to use it as parameter 'T' in the generic type or method 'MyList<T>'
+            //         MyList<int> list = [with(42)];
+            Diagnostic(ErrorCode.ERR_RefConstraintNotSatisfied, "int").WithArguments("MyList<T>", "T", "int").WithLocation(12, 16));
+    }
+
+    #endregion
+
+    #region Null and Default Tests
+
+    [Fact]
+    public void WithElement_NullArguments()
+    {
+        var source = """
+            using System;
+            using System.Collections.Generic;
+            
+            class MyList<T> : List<T>
+            {
+                public string Value { get; }
+                
+                public MyList(string value) : base()
+                {
+                    Value = value ?? "null";
+                }
+            }
+            
+            class C
+            {
+                static void Main()
+                {
+                    MyList<int> list = [with((string)null), 1];
+                    Console.WriteLine(list.Value);
+                }
+            }
+            """;
+
+        CompileAndVerify(source, expectedOutput: IncludeExpectedOutput("null"));
+    }
+
+    [Fact]
+    public void WithElement_DefaultArguments()
+    {
+        var source = """
+            using System;
+            using System.Collections.Generic;
+            
+            class MyList<T> : List<T>
+            {
+                public int Value { get; }
+                
+                public MyList(int value) : base()
+                {
+                    Value = value;
+                }
+            }
+            
+            class C
+            {
+                static void Main()
+                {
+                    MyList<int> list = [with(default(int)), 1];
+                    Console.WriteLine(list.Value);
+                }
+            }
+            """;
+
+        CompileAndVerify(source, expectedOutput: IncludeExpectedOutput("0"));
+    }
+
+    #endregion
+
+    #region Error Recovery Tests
+
+    [Fact]
+    public void WithElement_SyntaxError_MissingCloseParen()
+    {
+        var source = """
+            using System.Collections.Generic;
+            
+            class C
+            {
+                void M()
+                {
+                    List<int> list = [with(capacity: 10];
+                }
+            }
+            """;
+
+        CreateCompilation(source).VerifyDiagnostics(
```

Comment:
Consider verifying what information semantic model gives about the collection expression and `with` argument list.